src/buffer.cc

```diff
@@ -210,6 +210,18 @@ FormatBuffer* Buffer::AsFormatBuffer() {
   return static_cast<FormatBuffer*>(this);
 }
 
+Result Buffer::CopyTo(Buffer* buffer) const {
+  if (buffer->width_ != width_)
+    return Result("Buffer::CopyBaseFields() buffers have a different width");
+  if (buffer->height_ != height_)
+    return Result("Buffer::CopyBaseFields() buffers have a different height");
+  if (buffer->size_ != size_)
+    return Result("Buffer::CopyBaseFields() buffers have a different size");
+  buffer->values_.clear();
```

Comment:
nit: clear() isn't needed as the buffer will get removed when we assign it below.